pkg/threadas/threadas_test.go

```diff
@@ -0,0 +1,192 @@
+//go:build darwin
+// +build darwin
+
+package threadas
+
+import (
+	"runtime"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	timeout = 100 * time.Millisecond
+
+	// Hardcode the default macOS uid and gid for testing.
+	targetUid = 501
+	targetGid = 20
+)
+
+func TestThreadNoPermission(t *testing.T) {
+	t.Parallel()
+
+	if syscall.Getuid() == 0 {
+		t.Skip("Skipping -- test requires not-root")
+	}
+
+	uids := getUidData()
+	fn := uids.GenerateTestFunc(t, "expected failure", assert.Equal)
+	require.ErrorIs(t, ThreadAs(fn, timeout, uint32(uids.Uid), uint32(uids.Gid)), &NoPermissionsError{}, "Fails when run as non-root")
```

Comment:
this in conjunction with returning `fmt.Errorf()` from `ThreadAs()` will work and you can git rid of custom error
```suggestion
	require.ErrorIs(t, ThreadAs(fn, timeout, uint32(uids.Uid), uint32(uids.Gid)), syscall.Errno(1), "Fails when run as non-root")
```